c_api/src/c_api.rs

```diff
@@ -41,6 +41,28 @@ mod cbindgen_fix {
     pub struct C2paReader;
 }
 
+/// The location in which the manifest was read.
+#[repr(C)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum C2paManifestLocation {
+    /// The manifest is embedded in the asset.
+    Embedded,
+    /// The manifest is stored remotely.
+    Remote,
+    /// The manifest is stored separately in a sidecar.
+    Sidecar,
+}
+
```

Comment:
The only comment I have is can we change this from ManifestLocation to something else like ManifestDiscovery.  The reason is that in the internal code manifest location refer to its position in the asset